bundle/bundle_test.go

```diff
@@ -208,11 +208,65 @@ func TestValuesOrDefaults_Required(t *testing.T) {
 	// Example: It makes perfect sense for a user to specify --set minimum=0
 	// and in so doing meet the requirement that a value be specified.
 	vals["minimum"] = 0
-	res, err := ValuesOrDefaults(vals, b)
+	res, err := ValuesOrDefaults(vals, b, "install")
 	is.NoError(err)
 	is.Equal(0, res["minimum"])
 }
 
+func TestValuesOrDefaults_NotApplicableToAction(t *testing.T) {
+	is := assert.New(t)
+	vals := map[string]interface{}{
+		"param-with-default-and-override": true,
+	}
+	b := &Bundle{
+		Definitions: map[string]*definition.Schema{
+			"param-with-default-not-applicable": {
+				Type:    "string",
+				Default: "foo",
+			},
+			"required-param-not-applicable": {
+				Type: "string",
+			},
+			"param-with-default": {
+				Type:    "boolean",
+				Default: false,
+			},
+			"param-with-default-and-override": {
+				Type:    "boolean",
+				Default: false,
+			},
+		},
+		Parameters: map[string]Parameter{
+			"param-with-default-not-applicable": {
+				Definition: "param-with-default-not-applicable",
+				ApplyTo: []string{
+					"uninstall",
+				},
+			},
+			"required-param-not-applicable": {
+				Definition: "required-param-not-applicable",
+				Required:   true,
+				ApplyTo: []string{
+					"uninstall",
+				},
+			},
+			"param-with-default": {
+				Definition: "param-with-default",
+			},
+			"param-with-default-and-override": {
+				Definition: "param-with-default-and-override",
+			},
+		},
+	}
+
+	res, err := ValuesOrDefaults(vals, b, "install")
+	is.NoError(err)
```

Comment:
suggestion: I recommend using a require here so that it doesn't continue trying to check the remaining asserts when an error occurs.